src/ekat/ekat_workspace_impl.hpp

```diff
@@ -37,6 +37,41 @@ WorkspaceManager<T, D>::WorkspaceManager(int size, int max_used, TeamPolicy poli
   init(*this, m_max_ws_idx, m_max_used);
 }
 
+template <typename T, typename D>
+WorkspaceManager<T, D>::WorkspaceManager(T* data, int size, int max_used,
+                                         TeamPolicy policy, const double& overprov_factor) :
+  m_tu(policy, overprov_factor),
+  m_max_ws_idx(m_tu.get_num_ws_slots()),
+  m_reserve( (sizeof(T) > 2*sizeof(int)) ? 1 :
+             (2*sizeof(int) + sizeof(T) - 1)/sizeof(T) ),
+  m_size(size),
+  m_total(m_size + m_reserve),
+  m_max_used(max_used),
+#ifndef NDEBUG
+  m_num_used("Workspace.m_num_used", m_max_ws_idx),
+  m_high_water("Workspace.m_high_water", m_max_ws_idx),
+  m_active("Workspace.m_active", m_max_ws_idx, m_max_used),
+  m_curr_names("Workspace.m_curr_names", m_max_ws_idx, m_max_used, m_max_name_len),
+  m_all_names("Workspace.m_all_names", m_max_ws_idx, m_max_names, m_max_name_len),
+  // A name's index in m_all_names is used to index into m_counts
+  m_counts("Workspace.m_counts", m_max_ws_idx, m_max_names, 2),
+#endif
+  m_next_slot("Workspace.m_next_slot", m_pad_factor*m_max_ws_idx),
+  m_data(data, m_max_ws_idx, m_total * m_max_used)
```

Comment:
It looks like the two constructors are identical except for this line. I'm wondering if there's a way to use C++11's constructor delegation to get some code reuse here?

Reply:
We can have a private ctor that inits everything except `m_data`, and the other two call it. Since that ctor would be private, nobody would be able to call it (avoiding creating a WSM with uninited `m_data`).

Reply:
I agree, but I'm running into the issue the private constructor would have the exact same inputs as the original (non-data) constructor. Not sure how to get around this..

Reply:
Maybe a setup method? You'd have to give up initialization lists, and use assignments inside the setup method, but that's just a cosmetic difference in this case.